server/service/microsoft_mdm.go

```diff
@@ -572,6 +784,234 @@ func (svc *Service) GetMDMWindowsPolicyResponse(ctx context.Context, authToken s
 	return &policyMsg, nil
 }
 
+// GetMDMWindowsEnrollResponse returns a valid RequestSecurityTokenResponseCollection message
+// secTokenMsg is the RequestSecurityToken message
+// authToken is the base64 encoded binary security token
+func (svc *Service) GetMDMWindowsEnrollResponse(ctx context.Context, secTokenMsg *fleet.RequestSecurityToken, authToken string) (*fleet.RequestSecurityTokenResponseCollection, error) {
+	if len(authToken) == 0 {
+		return nil, fleet.NewInvalidArgumentError("enroll response", "authToken is empty")
+	}
+
+	// Validate the binary security token
+	err := svc.validateBinarySecurityToken(ctx, authToken)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "validate binary security token")
+	}
+
+	// Checking if this device is already MDM enrolled
+	err = svc.isWindowsDeviceAlreadyMDMEnrolled(ctx, secTokenMsg)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "device is already enrolled")
+	}
```

Comment:
can we skip this check? one thing we learned recently in Apple land is that devices can re-enroll without unenrolling first, for example:

1. Device gets reset to factory settings (or formatted or whatever)
2. Device is restored to a back-up (not sure if this applies in Windows, but probably?)
3. VM is restored to a previous state